### DISCUSSION TITLE
Is a user level install possible?

### DISCUSSION THREAD

**Discussion by four43** [jle64/gnome-pass-search-provider#39] [Ideas]
I was looking at implementing my own gnome search extension and I was curious if they are available to be installed in user directories, like ~/.local/share and get registered with gnome-search? It seems like these user specific ones should be living there, installable per user. Have you seen a way to do that? Thanks for the great example to follow btw!

**Comment by jle64** [comment]
If the [doc](https://developer.gnome.org/documentation/tutorials/search-provider.html) is to be believed, it seems that it is mandatory to put at least a file in /usr for registering the provider:
```
In order to register the search provider with GNOME Shell, you must provide a key/value file in $(datadir)/gnome-shell/search-providers for your provider.
```
There is no mention of support for XDG dirs so I don't think its possible to use ~/.local (unless the doc is lacking).

If you're looking to make it easy to install for users, I think that you can make search providers as Gnome Shell extensions (that is, in js) which can install in ~/.local (and are distributable on extensions.gnome.org). But if you want to use Python, AFAIK a pure ~/.local install is not possible.

**Comment by four43** [comment]
Thanks so much for the answer. It does seem like going that route might be better in my case. Flipping over to JS ins't a huge hurdle, especially with some examples out there.
